pkg/cmd/run.go

```diff
@@ -109,7 +116,12 @@ func RunMicroshift(cfg *config.MicroshiftConfig, flags *pflag.FlagSet) error {
 
 	klog.Infof("Starting MicroShift")
 
-	ctx, cancel := context.WithCancel(context.Background())
+	_, rotationDate, err := certchains.WhenToRotateAtEarliest(certChains)
+	if err != nil {
+		klog.Fatalf("failed to determine when to rotate certificates: %v", err)
+	}
+
+	ctx, cancel := context.WithDeadline(context.Background(), rotationDate)
```

Comment:
Although the implementation is very clean, this will give no indication of what happened on reboot.
Is there any way to get a message thrown when the context cancelation happens from this deadline?

I would consider adding here a TODO about that, or looking into getting this check added to https://github.com/openshift/microshift/tree/main/pkg/sysconfwatch may be?

This is a nit, a TODO would be enough for now.

Reply:
TODO+JIRA ticket so we don't forget

Reply:
Good idea, I'll create a JIRA for that